### PR TITLE
update the Crypto App Getting Started and code example

### PR DESCRIPTION
**Pull request by elof**
This updates the Stream Worker code and adds information for how to configure the app in the Macrometa console.

### CODE REVIEW

File: docs/apps/crypto-trading.md [hunk 1]
```diff
@@ -3,34 +3,46 @@ sidebar_position: 2
...
-|----------- |----------|----|-----------|-----------|
-|  `demo@macrometa.io` | `_system` | `crypto-trading-app` | [**Crypto Trading GUI**](https://macrometacorp.github.io/demo-crypto-trading-5g/) |
 
 :::note
 Stream Workers is currently an Enterprise only feature. 
 Please contact support@macrometa.com if you have any questions.
 :::
 
-**Stream App:**
+### Configuring the Crypto App
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
### Configure the Crypto App
```

File: docs/apps/crypto-trading.md [hunk 2]
```diff
@@ -3,34 +3,46 @@ sidebar_position: 2
 title: Crypto Trading Bot
 ---
 
-### On GDN
+### Crypto App details
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]: Headings should start at h2, not h3. Change throughout.

File: docs/apps/crypto-trading.md [hunk 3]
```diff
@@ -3,34 +3,46 @@ sidebar_position: 2
...
 
 :::note
 Stream Workers is currently an Enterprise only feature. 
 Please contact support@macrometa.com if you have any questions.
 :::
 
-**Stream App:**
+### Configuring the Crypto App
+
+1. If you haven't already, [signup](https://auth.paas.macrometa.io/signup) for a free Macrometa Playground account.
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
1. If you haven't already, [sign up for a free Macrometa Playground account](https://auth.paas.macrometa.io/signup). 
```

Comment (diana-macrometa) [reply docs/apps/crypto-trading.md]: "signup" is a noun
"sign up" is a verb

File: docs/apps/crypto-trading.md [hunk 4]
```diff
@@ -3,34 +3,46 @@ sidebar_position: 2
...
 Stream Workers is currently an Enterprise only feature. 
 Please contact support@macrometa.com if you have any questions.
 :::
 
-**Stream App:**
+### Configuring the Crypto App
+
+1. If you haven't already, [signup](https://auth.paas.macrometa.io/signup) for a free Macrometa Playground account. 
+
+2. Login into the [Macrometa Console](https://auth.paas.macrometa.io/) Go to the `Collections` tab and create a new `Doc` collection named `trades` with `collection stream enabled` checked.
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
2. Log in to the [Macrometa Console](https://auth.paas.macrometa.io/) Go to the Collections tab and create a new `Doc` collection named `trades` with `collection stream enabled` checked.
```

Comment (diana-macrometa) [reply docs/apps/crypto-trading.md]: This is actually three steps. Please break it up.

Also, "login" is a noun. "Log in" is a verb.

Comment (diana-macrometa) [reply docs/apps/crypto-trading.md]: Per the [Google Developer Style Guide](https://developers.google.com/style/ui-elements), UI items that you click are bold, not code style.

File: docs/apps/crypto-trading.md [hunk 5]
```diff
@@ -3,34 +3,46 @@ sidebar_position: 2
...
+
+6. Click publish.
+
+Once you the publish the Stream Worker, go over to our [live frontend](https://macrometacorp.github.io/tutorial-cryptotrading/) and use your Macrometa account credentials to log into your account. Select the location you deployed the Stream Worker to, and you will see the app come to life!
+
+### Stream Worker Code
 
 ```js
-@App:name("crypto-trading-app")
+@App:name("Crypto-Trading-App")
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md:29]: The name, description, and version appear automatically. They are not needed here.

File: docs/apps/crypto-trading.md [hunk 6]
```diff
@@ -3,34 +3,46 @@ sidebar_position: 2
...
 :::
 
-**Stream App:**
+### Configuring the Crypto App
+
+1. If you haven't already, [signup](https://auth.paas.macrometa.io/signup) for a free Macrometa Playground account. 
+
+2. Login into the [Macrometa Console](https://auth.paas.macrometa.io/) Go to the `Collections` tab and create a new `Doc` collection named `trades` with `collection stream enabled` checked.
+
+3. Open the `Stream Workers` tab and create a new `Stream Worker` named `Crypto-Trading-App`.
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
3. Click **Stream Workers**.
4. Click **New Stream Worker** named `Crypto-Trading-App`.
```

File: docs/apps/crypto-trading.md [hunk 7]
```diff
@@ -3,34 +3,46 @@ sidebar_position: 2
...
+
+4. Copy and past the code example into the editor.
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]: Link to the code example. Also, you should not have empty lines between numbered steps.

File: docs/apps/crypto-trading.md [hunk 8]
```diff
@@ -3,34 +3,46 @@ sidebar_position: 2
...
+
+5. Click save and enable on location that is closest to where you are located around the globe.
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
5. Click **Save** and enable on location that is closest to where you are located around the globe.
```

Comment (diana-macrometa) [reply docs/apps/crypto-trading.md]: We don't want to validate the code? Is that a best practice? Also, give more specific instructions for how to enable the location.

File: docs/apps/crypto-trading.md [hunk 9]
```diff
@@ -3,34 +3,46 @@ sidebar_position: 2
...
+
+6. Click publish.
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
6. Click **Publish**.
```

File: docs/apps/crypto-trading.md [hunk 10]
```diff
@@ -3,34 +3,46 @@ sidebar_position: 2
...
+
+Once you the publish the Stream Worker, go over to our [live frontend](https://macrometacorp.github.io/tutorial-cryptotrading/) and use your Macrometa account credentials to log into your account. Select the location you deployed the Stream Worker to, and you will see the app come to life!
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]: If the user is using their credentials, then why are the demo credentials at the top of this page?

File: docs/apps/crypto-trading.md [hunk 11]
```diff
@@ -3,34 +3,46 @@ sidebar_position: 2
...
+
+### Stream Worker Code
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
### Crypto App Stream Worker Code
```

Comment (diana-macrometa) [reply docs/apps/crypto-trading.md]: Try to use specific headings.

File: docs/apps/crypto-trading.md [hunk 12]
```diff
@@ -3,34 +3,46 @@ sidebar_position: 2
...
 -- The trigger
-define trigger CryptoTraderEventsTrigger at every 5 sec;
+CREATE TRIGGER CryptoTraderEventsTrigger WITH ( interval = 5 sec );
 
 /*
-This app reads every 5 seconds the close prices from Coinbase, Bitstamp and Bitflyer exchanges APIs.
+This app reads every 5 seconds the close prices FROM Coinbase, Bitstamp and Bitflyer exchanges APIs.
 Then it calculates the average prices within 10 events window and creates a "BUY/SELL" trading strategy.
 The close and average prices are stored in CryptoTraderQuotesAvgXXX streams 
 whereas the strategy is kept in trades collection.
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
The strategy is kept in trades collection.
```

File: docs/apps/crypto-trading.md [hunk 13]
```diff
@@ -3,34 +3,46 @@ sidebar_position: 2
...
 
 -- The trigger
-define trigger CryptoTraderEventsTrigger at every 5 sec;
+CREATE TRIGGER CryptoTraderEventsTrigger WITH ( interval = 5 sec );
 
 /*
-This app reads every 5 seconds the close prices from Coinbase, Bitstamp and Bitflyer exchanges APIs.
+This app reads every 5 seconds the close prices FROM Coinbase, Bitstamp and Bitflyer exchanges APIs.
 Then it calculates the average prices within 10 events window and creates a "BUY/SELL" trading strategy.
 The close and average prices are stored in CryptoTraderQuotesAvgXXX streams
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
The close and average prices are stored in CryptoTraderQuotesAvgXXX streams.
```

File: docs/apps/crypto-trading.md [hunk 14]
```diff
@@ -39,153 +51,163 @@ whereas the strategy is kept in trades collection.
 /**
 Testing the Stream Application:
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
Test the Stream Application:
```

File: docs/apps/crypto-trading.md [hunk 15]
```diff
@@ -39,153 +51,163 @@ whereas the strategy is kept in trades collection.
...
     1. Publish the app
+       
     2. Start the GUI against the same federation
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]: Not sure what this procedure here is for, didn't you give steps for this already? Either way, rewrite to remove "federation."

File: docs/apps/crypto-trading.md [hunk 16]
```diff
@@ -39,153 +51,163 @@ whereas the strategy is kept in trades collection.
...
 */
 
 -- Streams for the http call requests
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
-- Streams for the HTTP call requests
```

File: docs/apps/crypto-trading.md [hunk 17]
```diff
@@ -39,153 +51,163 @@ whereas the strategy is kept in trades collection.
...
+CREATE SINK UsdCryptoTraderRequestStream WITH (type='http-call', publisher.url='https://api.pro.coinbase.com/products/btc-usd/ticker', method='GET', headers="'User-Agent:c8cep'", sink.id='coinbase-ticker', map.type='json') (triggered_time string);
 
-@sink(type='http-call', publisher.url='https://api.bitflyer.com/v1/ticker',
-      method='GET', sink.id='bitflyer-ticker', @map(type='json'))
-define stream JpyCryptoTraderRequestStream (triggered_time string);
+CREATE SINK EurCryptoTraderRequestStream WITH (type='http-call', publisher.url='https://www.bitstamp.net/api/v2/ticker/btceur', method='GET', sink.id='bitstamp-ticker', map.type='json') (triggered_time string);
+
+CREATE SINK JpyCryptoTraderRequestStream WITH (type='http-call', publisher.url='https://api.bitflyer.com/v1/ticker', method='GET', sink.id='bitflyer-ticker', map.type='json') (triggered_time string);
 
 -- Streams for the http call responses
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
-- Streams for the HTTP call responses
```

Comment (diana-macrometa) [reply docs/apps/crypto-trading.md]: Capitalize acronyms. Hypertext transfer protocol.

**Review by diana-macrometa — CHANGES_REQUESTED** [review]
Good start. I made some suggestions.

**Comment by elof** [comment]
@diana-macrometa great, I accepted them all!

**Comment by diana-macrometa** [comment]
> @diana-macrometa great, I accepted them all!

Cool! Btw, if you click the circling arrows next to a reviewer's name, then GitHub notifies them that the PR is ready for review. In my case, I get a Slack.

File: docs/apps/crypto-trading.md [hunk 18]
```diff
@@ -3,189 +3,212 @@ sidebar_position: 2
 title: Crypto Trading Bot
 ---
 
-### On GDN
+### Crypto App details
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]: Please change all headings at this level to h2.

File: docs/apps/crypto-trading.md [hunk 19]
```diff
@@ -3,189 +3,212 @@ sidebar_position: 2
...
 |----------- |----------|----|-----------|-----------|
 |  `demo@macrometa.io` | `_system` | `crypto-trading-app` | [**Crypto Trading GUI**](https://macrometacorp.github.io/tutorial-cryptotrading/) |[github](https://github.com/Macrometacorp/tutorial-cryptotrading)|
 
-### On GDN (Wavelength 5G)
-
-| **Tenant** | **Fabric** | **Stream App** | **GUI** | **Source Code**|
-|----------- |----------|----|-----------|-----------|
-|  `demo@macrometa.io` | `_system` | `crypto-trading-app` | [**Crypto Trading GUI**](https://macrometacorp.github.io/demo-crypto-trading-5g/) |
 
 :::note
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]: Remove this note.

File: docs/apps/crypto-trading.md [hunk 20]
```diff
@@ -3,189 +3,212 @@ sidebar_position: 2
...
 Stream Workers is currently an Enterprise only feature. 
 Please contact support@macrometa.com if you have any questions.
 :::
 
-**Stream App:**
+### Configure the Crypto App
+
+1. If you haven't already, [sign up for a free Macrometa Playground account](https://auth.paas.macrometa.io/signup).
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]: Remove this step. It's covered in the following step for logging in. You might take a look at the tasks in the Streams or Collections sections to see how we do this.

File: docs/apps/crypto-trading.md [hunk 21]
```diff
@@ -3,189 +3,212 @@ sidebar_position: 2
...
 Stream Workers is currently an Enterprise only feature. 
 Please contact support@macrometa.com if you have any questions.
 :::
 
-**Stream App:**
+### Configure the Crypto App
+
+1. If you haven't already, [sign up for a free Macrometa Playground account](https://auth.paas.macrometa.io/signup). 
+
+2. Log in to the [Macrometa Console](https://auth.paas.macrometa.io/) Go to the Collections tab and create a new `Doc` collection named `trades` with `collection stream enabled` checked.
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]: Rewrite this as multiple steps. You can link to instructions for tasks like Create a Document Collection.

**Review by diana-macrometa — CHANGES_REQUESTED** [review]
@elof, please reread my comments. Accepting them does not help when I asked a question or requested a more complex change.

File: docs/apps/crypto-trading.md (hunk 10 above)
Comment (elof) [reply docs/apps/crypto-trading.md]: They can use the demo creds to see one in action without needed to set up their account.

**Comment by elof** [comment]
@diana-macrometa submitted my updates

File: docs/apps/crypto-trading.md [hunk 22]
```diff
@@ -3,178 +3,197 @@ sidebar_position: 2
...
 
-## On GDN
+## Crypto App details
 
 | **Tenant** | **Fabric** | **Stream App** | **GUI** | **Source Code**|
 |----------- |----------|----|-----------|-----------|
 |  `demo@macrometa.io` | `_system` | `crypto-trading-app` | [**Crypto Trading GUI**](https://macrometacorp.github.io/tutorial-cryptotrading/) |[github](https://github.com/Macrometacorp/tutorial-cryptotrading)|
 
-## Stream App
+Read our [Blog Post](https://www.macrometa.com/blog/cryptocurrency-trading-building-a-multi-exchange-global-trading-bot) about exchange arbitrage and how the app works.
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
Read our blog post [How To Build A Crypto Arbitrage Trading Bot](https://www.macrometa.com/blog/cryptocurrency-trading-building-a-multi-exchange-global-trading-bot) to learn more about exchange arbitrage and how the app works.
```

Comment (diana-macrometa) [reply docs/apps/crypto-trading.md]: "Blog post" is extremely vague link text. Try to be as specific as possible, preferably matching the title of whatever you are linking to.

File: docs/apps/crypto-trading.md [hunk 23]
```diff
@@ -3,178 +3,197 @@ sidebar_position: 2
...
+
+## Configure the Crypto App
+
+1. Log in to the [Macrometa Console](https://auth.paas.macrometa.io/) 
+2. Go to the Collections tab and create a new `Doc` collection named `trades` with `collection stream enabled` checked.
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
2. Go to the Collections tab and create a new document collection named `trades` with `collection stream enabled` checked. For more information about creating document collections, refer to [Create a Document Collection](ADD LINK TEXT HERE).
```

File: docs/apps/crypto-trading.md [hunk 24]
```diff
@@ -3,178 +3,197 @@ sidebar_position: 2
...
+3. Click **Stream Workers**.
+4. Click **New Stream Worker** named `Crypto-Trading-App`.
+5. Copy and past the [code example](#crypto-app-stream-worker-code) into the editor.
+6. Click **Save**. When you do we validate your code.
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
6. Click **Save**. 

    Macrometa validates your code.
```

File: docs/apps/crypto-trading.md [hunk 25]
```diff
@@ -3,178 +3,197 @@ sidebar_position: 2
...
+7. After you click save you will be prompted to select one or more locations to deploy your Stream Worker. Choose the location that is closest to where you are located, and click **Save**.
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
7. Select one or more locations to deploy your stream worker. Choose the location that is closest to where you are located, and then click **Save**.
```

File: docs/apps/crypto-trading.md [hunk 26]
```diff
@@ -3,178 +3,197 @@ sidebar_position: 2
...
+8. Click **Publish**.
+
+Once you the publish the Stream Worker, go over to our [live frontend](https://macrometacorp.github.io/tutorial-cryptotrading/) and use your Macrometa account credentials to log into your account. Select the location you deployed the Stream Worker to, and you will see the app come to life!
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
After you the publish the stream worker, navigate to the [live frontend](https://macrometacorp.github.io/tutorial-cryptotrading/) and use your Macrometa account credentials to log in to your account. Select the location you deployed the stream worker to, and you will see the app come to life!
```

File: docs/apps/crypto-trading.md (hunk 24 above)
Comment (diana-macrometa) [reply docs/apps/crypto-trading.md]: Can you add a little more here about the validation? What if it says the code is invalid? Is the validate line even useful in this context, or should we take it out?

**Review by diana-macrometa — CHANGES_REQUESTED** [review]
Almost there!

File: docs/apps/crypto-trading.md [hunk 27]
```diff
@@ -3,178 +3,196 @@ sidebar_position: 2
 title: Crypto Trading Bot
 ---
 
-## On GDN
+## Crypto App details
 
 | **Tenant** | **Fabric** | **Stream App** | **GUI** | **Source Code**|
 |----------- |----------|----|-----------|-----------|
 |  `demo@macrometa.io` | `_system` | `crypto-trading-app` | [**Crypto Trading GUI**](https://macrometacorp.github.io/tutorial-cryptotrading/) |[github](https://github.com/Macrometacorp/tutorial-cryptotrading)|
```
Comment (diana-macrometa) [inline docs/apps/crypto-trading.md]:
```suggestion
|  `demo@macrometa.io` | `_system` | `Crypto-Trading-App` | [**Crypto Trading GUI**](https://macrometacorp.github.io/tutorial-cryptotrading/) |[github](https://github.com/Macrometacorp/tutorial-cryptotrading)|
```

**Review by diana-macrometa — APPROVED** [review]
One minor suggestion, then good to go!

**Comment by rashimak** [comment]
I need a Crypto trading bot

**Pull request merged**